ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -20,11 +22,30 @@
     (v/validate-concept-metadata concept)
     concept))
 
+(defn- check-subscriber-collection-permission
+  [request-context concept]
+  (let [metadata (-> (:metadata concept)
+                     (json/decode true))
+        concept-id (:CollectionConceptId metadata)
+        subscriber-id (:SubscriberId metadata)
+        permissions (-> (access-control/get-permissions request-context
+                                                        {:concept_id concept-id
+                                                         :user_id subscriber-id})
+                        json/decode
+                        (get concept-id))]
+    (when-not (some #{"read"} permissions)
+      (errors/throw-service-error
+       :unauthorized
+       (format "Subscriber-id [%s] does not have permission to view collection [%s]."
+               subscriber-id
+               concept-id)))))
+
```

Comment:
I remember this was a pain point before that we can't just check the permission on the collection because the subscription is really on the granules.  and the permission on the granules are not guaranteed just because the permission on the collection is granted.

Reply:
This isn't a problem.